client/webserver/site/src/js/order.ts

```diff
@@ -258,33 +263,86 @@ export default class OrderPage extends BasePage {
 
-    const setCoin = (pendingName: string, linkName: string, coin: Coin) => {
-      const formatCoinID = (cid: string) => {
-        if (cid.startsWith(coinIDTakerFoundMakerRedemption)) {
-          const makerAddr = cid.substring(coinIDTakerFoundMakerRedemption.length)
-          return intl.prep(intl.ID_TAKER_FOUND_MAKER_REDEMPTION, { makerAddr: makerAddr })
-        }
-        return cid
-      }
+    const tryShowCoin = (showPlaceholder: () => void, showCoin: () => void, linkName: string, coin: Coin) => {
       const coinLink = tmpl[linkName]
-      const pendingSpan = tmpl[pendingName]
       if (!coin) {
-        Doc.show(tmpl[pendingName])
-        Doc.hide(tmpl[linkName])
+        Doc.hide(coinLink)
+        showPlaceholder()
         return
       }
       coinLink.textContent = formatCoinID(coin.stringID)
       coinLink.dataset.explorerCoin = coin.stringID
       setCoinHref(coin.assetID, coinLink)
-      Doc.hide(pendingSpan)
       Doc.show(coinLink)
+      showCoin()
     }
 
-    setCoin('makerSwapPending', 'makerSwapCoin', makerSwapCoin(m))
-    setCoin('takerSwapPending', 'takerSwapCoin', takerSwapCoin(m))
-    setCoin('makerRedeemPending', 'makerRedeemCoin', makerRedeemCoin(m))
-    setCoin('takerRedeemPending', 'takerRedeemCoin', takerRedeemCoin(m))
-    setCoin('refundPending', 'refundCoin', m.refund)
+    tryShowCoin(
+      (): void => {
+        Doc.show(tmpl.makerSwapPending)
+      },
+      (): void => {
+        Doc.hide(tmpl.makerSwapPending)
+      },
+      'makerSwapCoin',
+      makerSwapCoin(m)
+    )
+    tryShowCoin(
+      (): void => {
+        Doc.show(tmpl.takerSwapPending)
+      },
+      (): void => {
+        Doc.hide(tmpl.takerSwapPending)
+      },
+      'takerSwapCoin',
+      takerSwapCoin(m)
+    )
+    tryShowCoin(
+      (): void => {
+        Doc.show(tmpl.makerRedeemPending)
+      },
+      (): void => {
+        Doc.hide(tmpl.makerRedeemPending)
+      },
+      'makerRedeemCoin',
+      makerRedeemCoin(m)
+    )
+    tryShowCoin(
+      (): void => {
+        Doc.show(tmpl.takerRedeemPending)
+      },
+      (): void => {
+        Doc.hide(tmpl.takerRedeemPending)
+      },
+      'takerRedeemCoin',
+      takerRedeemCoin(m)
+    )
+    tryShowCoin(
+      (): void => {
+        let lockTime = lockTimeMakerMs
+        if (m.side === OrderUtil.Taker) {
+          lockTime = lockTimeTakerMs
+        }
+        const refundAfter = new Date(m.stamp + lockTime)
+        if (Date.now() > refundAfter.getTime()) {
+          tmpl.refundPending.textContent = intl.prep(intl.ID_REFUND_COMING_YOUR_WAY)
+        } else {
+          const refundAfterStr = refundAfter.toLocaleTimeString('en-GB', {
+            year: 'numeric',
+            month: 'short',
+            day: 'numeric'
+          })
+          tmpl.refundPending.textContent = intl.prep(intl.ID_REFUND_WILL_HAPPEN_AFTER, { refundAfterTime: refundAfterStr })
+        }
+
+        Doc.show(tmpl.refundPending)
+      },
```

Comment:
`setCoin` func (renamed to `tryShowCoin` now) refactoring in this PR is mostly to accommodate this change here ^